modin/data_management/partitioning/partition_collections.py

```diff
@@ -0,0 +1,974 @@
+            chunk_builder(i, j)
+            for j in range(0, len(df.columns), col_chunksize)
+        ] for i in range(0, len(df), row_chunksize)]
+
+        return cls(np.array(parts))
+
+    def get_indices(self, axis=0, index_func=None, old_blocks=None):
+        """This gets the internal indices stored in the partitions.
+
+        Note: These are the global indices of the object. This is mostly useful
+            when you have deleted rows/columns internally, but do not know
+            which ones were deleted.
+
+        Args:
+            axis: This axis to extract the labels. (0 - index, 1 - columns).
+            index_func: The function to be used to extract the function.
+            scale_index: True if we need to add the lengths of previous blocks,
+                False otherwise. This should be True if each block had the
+                index reset based on the length of that particular block,
+                otherwise it should be False.
+            old_blocks: An optional previous object that this object was
+                created from. This is used to compute the correct offsets.
+
+        Returns:
+            A Pandas Index object.
+        """
+        assert callable(index_func), \
+            "Must tell this function how to extract index"
+
+        if axis == 0:
+            func = self.preprocess_func(index_func)
+            # We grab the first column of blocks and extract the indices
+            new_indices = [
+                idx.apply(func).get() for idx in self.partitions.T[0]
+            ]
+            # This is important because sometimes we have resized the data. The new
+            # sizes will not be valid if we are trying to compute the index on a
+            # new object that has a different length.
+            if old_blocks is not None:
+                cumulative_block_lengths = np.array(
+                    old_blocks.block_lengths).cumsum()
+            else:
+                cumulative_block_lengths = np.array(
+                    self.block_lengths).cumsum()
+        else:
+            func = self.preprocess_func(index_func)
+            new_indices = [idx.apply(func).get() for idx in self.partitions[0]]
+
+            if old_blocks is not None:
+                cumulative_block_lengths = np.array(
+                    old_blocks.block_widths).cumsum()
+            else:
+                cumulative_block_lengths = np.array(self.block_widths).cumsum()
+
+        full_indices = new_indices[0]
+
+        if old_blocks is not None:
+            for i in range(len(new_indices)):
+                # If the length is 0 there is nothing to append.
+                if i == 0 or len(new_indices[i]) == 0:
+                    continue
+                # The try-except here is intended to catch issues where we are
+                # trying to get a string index out of the internal index.
+                try:
+                    append_val = new_indices[i] + cumulative_block_lengths[i -
```

Comment:
is it necessarily adding old block length here?

Reply:
Yes, we need this to scale the index properly.

Internal indices are `RangeIndex(0, len(partition)`. Scaling gives us a global index.